cloud/amazon/s3.py

```diff
@@ -409,6 +412,10 @@ def main():
     s3_url = module.params.get('s3_url')
     src = module.params.get('src')
 
+    for acl in module.params.get('permission').split(','):
```

Comment:
make permission a list and you won't have do do this, also you don't check for , first so this can cause an exception.